api/info/connecthandler/info_connect.go

```diff
@@ -0,0 +1,325 @@
+		nodePOP = make([]byte, len(p.PublicKey)+len(p.ProofOfPossession))
+		copy(nodePOP[0:len(p.PublicKey)], p.PublicKey[:])
+		copy(nodePOP[len(p.PublicKey):], p.ProofOfPossession[:])
+	}
+
+	response := &v1.GetNodeIDResponse{
+		NodeId:  jsonResponse.NodeID.String(),
+		NodePop: nodePOP,
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetNodeIP returns the primary IP address this node uses for P2P networking.\
+func (c *connectInfoService) GetNodeIP(
+	_ context.Context,
+	_ *connect.Request[v1.GetNodeIPRequest],
+) (*connect.Response[v1.GetNodeIPResponse], error) {
+	var jsonResponse info.GetNodeIPReply
+	if err := c.Info.GetNodeIP(nil, nil, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetNodeIPResponse{
+		Ip: jsonResponse.IP.String(),
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetNetworkID returns the numeric ID of the Avalanche network this node is connected to
+func (c *connectInfoService) GetNetworkID(
+	_ context.Context,
+	_ *connect.Request[v1.GetNetworkIDRequest],
+) (*connect.Response[v1.GetNetworkIDResponse], error) {
+	var jsonResponse info.GetNetworkIDReply
+	if err := c.Info.GetNetworkID(nil, nil, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetNetworkIDResponse{
+		NetworkId: uint32(jsonResponse.NetworkID),
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetNetworkName returns the name of the network
+func (c *connectInfoService) GetNetworkName(
+	_ context.Context,
+	_ *connect.Request[v1.GetNetworkNameRequest],
+) (*connect.Response[v1.GetNetworkNameResponse], error) {
+	var jsonResponse info.GetNetworkNameReply
+	if err := c.Info.GetNetworkName(nil, nil, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetNetworkNameResponse{
+		NetworkName: jsonResponse.NetworkName,
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetBlockchainID maps an ID string to its canonical chain ID
+func (c *connectInfoService) GetBlockchainID(
+	_ context.Context,
+	request *connect.Request[v1.GetBlockchainIDRequest],
+) (*connect.Response[v1.GetBlockchainIDResponse], error) {
+	jsonRequest := info.GetBlockchainIDArgs{
+		Alias: request.Msg.Alias,
+	}
+
+	var jsonResponse info.GetBlockchainIDReply
+	if err := c.Info.GetBlockchainID(nil, &jsonRequest, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetBlockchainIDResponse{
+		BlockchainId: jsonResponse.BlockchainID.String(),
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// Peers returns metadata (IP, nodeID, version, uptimes, subnets, etc.) for the given peer node IDs
+func (c *connectInfoService) Peers(
+	_ context.Context,
+	request *connect.Request[v1.PeersRequest],
+) (*connect.Response[v1.PeersResponse], error) {
+	nodeIDs := make([]ids.NodeID, 0, len(request.Msg.NodeIds))
+	for _, nodeIDStr := range request.Msg.NodeIds {
+		nodeID, err := ids.NodeIDFromString(nodeIDStr)
+		if err != nil {
+			return nil, connect.NewError(
```

Comment:
Can we just return this error and not worry about error codes (so connect will just use the default unspecified code)? I'm open to introducing new error codes as they're needed but for now without a clear reason for them I think it's a bit premature.